src/dbnode/storage/index/convert/convert_test.go

```diff
@@ -68,37 +68,85 @@ func TestFromSeriesIDAndTagIteratorInvalid(t *testing.T) {
 }
 
 func TestFromSeriesIDAndTagsValid(t *testing.T) {
-	id := ident.StringID("foo")
+	tests := []struct {
+		name string
+		id   string
+	}{
+		{
+			name: "no tags in ID",
+			id:   "foo",
+		},
+		{
+			name: "tags in ID",
+			id:   "bar=baz",
+		},
+		{
+			name: "tags in ID with specific format",
+			id:   `{bar="baz"}`,
+		},
+		{
+			name: "inexact tag occurrence in ID",
+			id:   "bazillion_barometers",
+		},
+	}
```

Comment:
I think we need more test cases, also covering the multi-tag conversion. You have the logics that predicts the location of the second tag based on the end of the first tag, I believe it is not covered.
Also, the goal is to avoid allocation, so it would be beneficial to verify that tag slices actually come from the id when possible. This is not trivial, but I think you can come up with something, like mutating a shared value and checking that it changed on the other slice.

Reply:
Can probably find a way to do it by using this: https://github.com/m3db/m3/blob/ac2ef9b9da9685cfdad2312aca35bf093c1ebb38/src/x/test/util.go#L30